scripts/feature-gen.py

```diff
@@ -0,0 +1,249 @@
+    if 'operationId' not in spec[url_path][http_method]:
+        raise ValueError('operationId must be present in spec.yaml')
+    return spec[url_path][http_method]['operationId']
+
+def format_url(url_path):
+    return "f'" + url_path + "'"
+
+def build_validations(propName, propConfig, required_props, validations):
+    if propConfig['type'] == 'string' and 'minLength' in propConfig:
+        if propName in required_props:
+            validations.append((f'if len({propName}) < {propConfig["minLength"]}', f'raise PineconeApiValueError("{propName} must be at least {propConfig["minLength"]} characters long")'))
+        else:
+            validations.append((f'if {propName} is not None and len({propName}) < {propConfig["minLength"]}', f'raise PineconeApiValueError("{propName} must be at least {propConfig["minLength"]} characters long")'))
+
+    if propConfig['type'] == 'string' and 'maxLength' in propConfig:
+        if propName in required_props:
+            validations.append((f'if len({propName}) > {propConfig["maxLength"]}', f'raise PineconeApiValueError("{propName} must be at most {propConfig["maxLength"]} characters long")'))
+        else:
+            validations.append((f'if {propName} is not None and len({propName}) > {propConfig["maxLength"]}', f'raise PineconeApiValueError("{propName} must be at most {propConfig["maxLength"]} characters long")'))
+
+    if propConfig['type'] == 'integer' and 'minimum' in propConfig:
+        if propName in required_props:
+            validations.append((f'if {propName} < {propConfig["minimum"]}', f'raise PineconeApiValueError("{propName} must be at least {propConfig["minimum"]}")'))
+        else:
+            validations.append((f'if {propName} is not None and {propName} < {propConfig["minimum"]}', f'raise PineconeApiValueError("{propName} must be at least {propConfig["minimum"]}")'))
+
+    if propConfig['type'] == 'integer' and 'maximum' in propConfig:
+        if propName in required_props:
+            validations.append((f'if {propName} > {propConfig["maximum"]}', f'raise PineconeApiValueError("{propName} must be at most {propConfig["maximum"]}")'))
+        else:
+            validations.append((f'if {propName} is not None and {propName} > {propConfig["maximum"]}', f'raise PineconeApiValueError("{propName} must be at most {propConfig["maximum"]}")'))
+
+    if propConfig['type'] == 'string' and 'enum' in propConfig:
+        if propName in required_props:
+            validations.append((f'if {propName} not in {propConfig["enum"]}', f'raise PineconeApiValueError("{propName} must be one of {propConfig["enum"]}")'))
+        else:
+            validations.append((f'if {propName} is not None and {propName} not in {propConfig["enum"]}', f'raise PineconeApiValueError("{propName} must be one of {propConfig["enum"]}")'))
+
+type_translation = {
+    'string': 'str',
+    'integer': 'int',
+    'boolean': 'bool',
+    'array': 'List',
+    'object': 'Dict',
+    'number': 'float'
```

Comment:
OOC: why do you do `List` (instead of `list`) and `Dict` instead of (`dict`) here?

Reply:
(Mostly ask b/c I think [Python's moving towards](https://docs.python.org/3/library/typing.html#typing.List) deprecating the `typing.<whatever>` types in favor of built-ins)